ui/apps/platform/src/services/NetworkService.js

```diff
@@ -346,14 +347,25 @@ export function getActiveNetworkModification(clusterId, deploymentQuery) {
     });
 }
 
-export function fetchNetworkPoliciesByClusterId(clusterId) {
-    const params = queryString.stringify({ clusterId });
+/**
+ * Fetches the network policies applied to deployments in the given scope.
+ * @param {!String} clusterId The cluster ID.
+ * @param {!String} deploymentQuery A search filter string.
+ * @returns {Promise<import("../types/networkPolicy.proto").NetworkPolicy[]>}
+ */
+export function fetchNetworkPoliciesByClusterId(clusterId, deploymentQuery) {
+    if (clusterId === '') {
+        return Promise.resolve([]);
+    }
+    // The `deploymentQuery` param functions identically to the general `query` param used in
+    // other API calls and accepts the same search filter syntax.
+    const params = queryString.stringify({ clusterId, deploymentQuery });
     const options = {
         method: 'GET',
         url: `${networkPoliciesBaseUrl}?${params}`,
     };
     return axios(options).then((response) => {
-        return response?.data?.networkPolicies;
+        return sortBy(response?.data?.networkPolicies, 'name') ?? [];
```

Comment:
What are pro and con
* encapsulate `sortBy` in service function
* let caller take responsibility, especially if call is encapsulated in hook?

Reply:
Good idea. I'm moving some of this code around in a follow up and made a note to extract the sorting in my current branch.